### PR TITLE
Implement project permissions tests #LMR-276

### PR DESCRIPTION
**Pull request by SquattingPenguin**

### CODE REVIEW

File: war/src/main/java/io/lumeer/engine/controller/ProjectFacade.java [hunk 1]
```diff
@@ -117,15 +117,18 @@ public void setCurrentProjectCode(final String projectCode) {
...
     */
    public List<Project> readProjects(final String organizationCode) {
-      return dataStorage.search(LumeerConst.Project.COLLECTION_NAME, organizationIdFilter(organizationFacade.getOrganizationId(organizationCode)), null, 0, 0)
-                        .stream().map(Project::new).collect(Collectors.toList());
+      return dataStorage.search(LumeerConst.Project
+            .COLLECTION_NAME, organizationIdFilter(organizationFacade.getOrganizationId(organizationCode)), null, 0, 0)
+                        .stream().map(Project::new)
+                        .filter(o -> securityFacade.hasProjectRole(o.getCode(), LumeerConst.Security.ROLE_READ))
+                        .collect(Collectors.toList());
    }
```
Comment (kubedo8) [inline war/src/main/java/io/lumeer/engine/controller/ProjectFacade.java]: Security checks should be implemented in service layer.

Comment (SquattingPenguin) [inline war/src/main/java/io/lumeer/engine/controller/ProjectFacade.java]: @kubedo8 It is implemented in the same way in OrganizationFacade already.

File: war/src/test/java/io/lumeer/engine/rest/permissions/ProjectServicePermissionsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
+package io.lumeer.engine.rest.permissions;
```
Comment (livthomas) [inline war/src/test/java/io/lumeer/engine/rest/permissions/ProjectServicePermissionsIntegrationTest.java]: License header is missing.

File: war/src/test/java/io/lumeer/engine/rest/permissions/ProjectServicePermissionsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      createClient();
+      String newprojectCode = "NewCode";
+      String newProjectName2 = "NewName2";
+      response = client.target(TARGET_URI).path(PATH_PREFIX + projectCode).
+            request(MediaType.APPLICATION_JSON).buildPut(Entity.json(new Project(newprojectCode, newProjectName2))).invoke();
+      assertThat(response.getStatusInfo()).isEqualTo(Response.Status.UNAUTHORIZED);
+   }
+
+   @Test
+   public void testGetUpdateProjectManageRole() throws UserAlreadyExistsException {
```
Comment (livthomas) [inline war/src/test/java/io/lumeer/engine/rest/permissions/ProjectServicePermissionsIntegrationTest.java]: This is probably a type and `Get` should be removed from the method name, or not?

**Pull request merged**